src/homework8/arraylist/ArrayList.java

```diff
@@ -0,0 +1,107 @@
+package homework8.arraylist;
+
+import java.util.Arrays;
+
+public class ArrayList<T> implements List<T> {
+    private int position = 0;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[10];
+    }
+
+    @Override
+    public void add(T value) {
+        if (position < array.length) {
+            array[position] = value;
+            position++;
+        } else {
+            int newArrayLength = array.length + (array.length >> 1);
+            array = Arrays.copyOf(array, newArrayLength);
+            array[position] = value;
+            position++;
+        }
+    }
+
+    @Override
+    public void add(T value, int index) {
+        try {
```

Comment:
better add checking:
if (position >= array.length) 

and checking to index: if (index>position)